demo/midgard-node/src/transactions/utils.ts

```diff
@@ -21,26 +21,47 @@ import { Database } from "@/services/database.js";
 export const handleSignSubmit = (
   lucid: LucidEvolution,
   signBuilder: TxSignBuilder,
-): Effect.Effect<string, Error> =>
+  onSubmitFailure: (error: SubmitError) => Effect.Effect<void>,
+  onConfirmFailure: (error: ConfirmError) => Effect.Effect<void>,
```

Comment:
```suggestion
  onSubmitFailure: (error: SubmitError) => Effect.Effect<void, Error>,
  onConfirmFailure: (error: ConfirmError) => Effect.Effect<void, Error>,
```
I'm not entirely sure if this is needed, but lets be as explicit as we can.